app/controllers/main_controller.rb

```diff
@@ -1,9 +1,36 @@
 class MainController < ApplicationController
-  def home
-    if current_user
-      redirect_to exams_path
-    else
+  def index
+    unless current_user
       redirect_to new_user_session_path
+      return
     end
+
+    registrations = current_user.registrations
+    exams =
+      registrations
+      .map do |r|
+        e = r.exam
+        return nil unless e.enabled?
```

Comment:
This `return` will bail all the way out of `index`, not just this loop iteration.